satpy/readers/viirs_sdr.py

```diff
@@ -515,13 +231,23 @@ def filter_filenames_by_info(self, filename_items):
                     geo_keep.append(filename)
         if geo_keep:
             filename_items = self._remove_geo_datasets_from_files(filename_items, geo_del)
-        return super(VIIRSSDRReader, self).filter_filenames_by_info(filename_items)
+
+        filename_items = self._remove_non_viirs_datasets_from_files(filename_items, viirs_del)
+        return super().filter_filenames_by_info(filename_items)
+
+    def _remove_non_viirs_datasets_from_files(self, filename_items, files_to_edit):
+        no_viirs = ATMS_DATASET_KEYS
+        return self._remove_datasets_from_files(filename_items, files_to_edit, no_viirs)
```

Comment:
Na, the thing is that there may be ATMS SDR files in the same directory, and the names are following exactly same naming convention. This means the patterns specified in the `viirs_sdr.yaml` will also pick up the ATMS files, e.g.:
```
GATMO_j01_d20221220_t0910240_e0921356_b26361_c20221220100456680030_cspp_dev.h5
SATMS_j01_d20221220_t0910240_e0921356_b26361_c20221220100456348770_cspp_dev.h5
TATMS_j01_d20221220_t0910240_e0921356_b26361_c20221220100456524427_cspp_dev.h5
```
The file patterns specified in the yaml file is like this:
```
    file_patterns: ['{datasets}_{platform_shortname}_d{start_time:%Y%m%d_t%H%M%S%f}_e{end_time:%H%M%S%f}_b{orbit:5d}_c{creation_time:%Y%m%d%H%M%S%f}_{source}.h5']
```

In the atms counterpart it is much easier to be more specific as there are only three types of files (as above). For VIIRS there are many, so I think it would be cumbersome to specify this full list in the yaml file?

Reply:
yes, it would be cumbersome, especially since some viirs files can contain multiple channels and thus have highly variable names... The only worry then is what happens when there are both atms and viirs files in the same directory, to they get assigned to the readers correctly? Or should the viirs reader raise some kind of error when being assigned an atms file?

Reply:
I have added a test for the case of both ATMS and VIIRS SDRs in the same directory. See below.
I think that should be enough?